### PR TITLE
Ensure that all `.jld2`/`.nc` files are deleted before docs deployment

### PR DESCRIPTION
**Pull request by navidcy**
See below for a MWE.

```Julia
$ julia --project=docs
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.9.0 (2023-05-07)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

julia> using Glob

julia> glob("src/*.jl")
6-element Vector{String}:
 "src/Architectures.jl"
 "src/Biogeochemistry.jl"
 "src/Logger.jl"
 "src/Oceananigans.jl"
 "src/StokesDrift.jl"
 "src/Units.jl"

julia> glob("src/**/*.jl")
213-element Vector{String}:
 "src/AbstractOperations/AbstractOperations.jl"
 "src/AbstractOperations/at.jl"
 "src/AbstractOperations/binary_operations.jl"
 "src/AbstractOperations/broadcasting_abstract_operations.jl"
 "src/AbstractOperations/computed_field.jl"
 "src/AbstractOperations/conditional_operations.jl"
 "src/AbstractOperations/constant_field_abstract_operations.jl"
 "src/AbstractOperations/derivatives.jl"
 "src/AbstractOperations/grid_metrics.jl"
 "src/AbstractOperations/grid_validation.jl"
 "src/AbstractOperations/kernel_function_operation.jl"
 "src/AbstractOperations/metric_field_reductions.jl"
 "src/AbstractOperations/multiary_operations.jl"
 "src/AbstractOperations/show_abstract_operations.jl"
 "src/AbstractOperations/unary_operations.jl"
 "src/Advection/Advection.jl"
 "src/Advection/cell_advection_timescale.jl"
 "src/Advection/centered_advective_fluxes.jl"
 "src/Advection/centered_reconstruction.jl"
 "src/Advection/flat_advective_fluxes.jl"
 ⋮
 "src/TurbulenceClosures/implicit_explicit_time_discretization.jl"
 "src/TurbulenceClosures/isopycnal_rotation_tensor_components.jl"
 "src/TurbulenceClosures/turbulence_closure_diagnostics.jl"
 "src/TurbulenceClosures/turbulence_closure_utils.jl"
 "src/TurbulenceClosures/velocity_tracer_gradients.jl"
 "src/TurbulenceClosures/vertically_implicit_diffusion_solver.jl"
 "src/Utils/Utils.jl"
 "src/Utils/kernel_launching.jl"
 "src/Utils/multi_region_transformation.jl"
 "src/Utils/ordered_dict_show.jl"
 "src/Utils/output_writer_diagnostic_utils.jl"
 "src/Utils/pretty_filesize.jl"
 "src/Utils/prettysummary.jl"
 "src/Utils/prettytime.jl"
 "src/Utils/schedules.jl"
 "src/Utils/tuple_utils.jl"
 "src/Utils/user_function_arguments.jl"
 "src/Utils/versioninfo.jl"
 "src/Utils/with_tracers.jl"
```

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -160,9 +160,9 @@ makedocs(bib,
  checkdocs = :exports
 )
 
-@info "Cleaning up temporary .jld2 and .nc files created by doctests..."
+@info "Cleaning up temporary .jld2 and .nc output created by doctests or literated examples..."
 
-for file in vcat(glob("docs/*.jld2"), glob("docs/*.nc"))
+for file in vcat(glob("docs/**/*.jld2"), glob("docs/**/*.nc"))
     rm(file)
 end
```
[the remark this answers is not in the document]
Comment (navidcy) [reply docs/make.jl:167]: I’ll post something here in a bit

[the remark this answers is not in the document]
Comment (navidcy) [reply docs/make.jl:167]: I put a mew in the first post of this PR.

**Comment by tomchor** [comment]
I think replacing `"docs/*.jld2"` with `"docs/**/*.jld2"` misses the stuff that previously would have been caught:

```julia
    ~/repos/Oceananigans.jl    tc/nhpressure2 *2 !7 ?1    julia --project=docs/                                                                                    ✔  9s   base  
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.5 (2023-01-08)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Glob

julia> A = glob("src/*.jl")
6-element Vector{String}:
 "src/Architectures.jl"
 "src/Biogeochemistry.jl"
 "src/Logger.jl"
 "src/Oceananigans.jl"
 "src/StokesDrift.jl"
 "src/Units.jl"

julia> B = glob("src/**/*.jl");

julia> A[1] in B
false

julia> collect( a in B for a in A )
6-element Vector{Bool}:
 0
 0
 0
 0
 0
 0
```

File: docs/make.jl [hunk 2]
```diff
@@ -160,9 +160,9 @@ makedocs(bib,
  checkdocs = :exports
 )
 
-@info "Cleaning up temporary .jld2 and .nc files created by doctests..."
+@info "Cleaning up temporary .jld2 and .nc output created by doctests or literated examples..."
 
-for file in vcat(glob("docs/*.jld2"), glob("docs/*.nc"))
+for file in vcat(glob("docs/**/*.jld2"), glob("docs/**/*.nc"))
```
Comment (tomchor) [inline docs/make.jl:165]: Maybe

```suggestion
for file in vcat(glob("docs/**/*.jld2"), glob("docs/**/*.nc"), glob("docs/*.jld2"), glob("docs/*.nc"))
```

?

Comment (navidcy) [reply docs/make.jl:165]: Hm... then we need to figure out a way to include all subdirectories recursively

Comment (navidcy) [reply docs/make.jl]: for now I added the directories that output lives in... It's not the best solution but it's better than uploading `.nc` files in the Docs repository!!

**Review by tomchor — APPROVED** [review]

**Pull request merged**